### PR TITLE
Fix issues 21635 and 21636 - check parameters of bitfields

### PR DESCRIPTION
**Pull request by berni44**
Additionally I fixed the old style using ".result" instead of template name and removed a superfluous import.

### CODE REVIEW

**Review by dukc — APPROVED** [review]
LGTM

File: std/bitmanip.d [hunk 1]
```diff
@@ -256,10 +255,42 @@ bool), followed by unsigned types, followed by signed types.
 
 See_Also: $(REF BitFlags, std,typecons)
 */
-
-template bitfields(T...)
+string bitfields(T...)()
 {
-    enum { bitfields = createStorageAndFields!T.result }
+    import std.traits : isIntegral, isSomeString;
```
Comment (berni44) [inline std/bitmanip.d:260]: Ooop, looks like I forgot to remove these imports...

**Comment by RazvanN7** [comment]
You need to also add the number for issue 21636 in the commit if you want the bot to pick it up.

**Comment by berni44** [comment]
> You need to also add the number for issue 21636 in the commit if you want the bot to pick it up.

That's strange. @Geod24 claimed too that 21636 isn't in the commit message. But it is there:

```
Fix issue 21635 - std.bitmanip: bitfields should produce better error messages with wrong parameters
Fix issue 21636 - std.bitmanip: bitfields size of bitfield should be checked against size of used type
```

Probably the bot didn't catch it, because it's in the second line?

**Comment by PetarKirov** [comment]
@berni44 You can find the details how dlang-bot parses the commit messages here: https://github.com/dlang/dlang-bot#referencing-multiple-issues

**Comment by berni44** [comment]
> @berni44 You can find the details how dlang-bot parses the commit messages here: https://github.com/dlang/dlang-bot#referencing-multiple-issues

Oh, thanks. There were two misunderstandings on my side: First I thought, the bot will also parse the name of the PR to find out about bugs. As it seems, it doesn't. Second, I did not know, that the whole commit message is parsed in one stroke. I thought this would be done line by line. Anyway, I changed the commit message now.

File: std/bitmanip.d [hunk 2]
```diff
@@ -282,10 +281,41 @@ Returns: A string that can be used in a `mixin` to add the `bitfield`.
...
 */
-
-template bitfields(T...)
+string bitfields(T...)()
 {
-    enum { bitfields = createStorageAndFields!T.result }
+    import std.conv : to;
+
+    static assert(T.length % 3 == 0,
+                  "Wrong number of arguments (" ~ to!string(T.length) ~ "): Must be dividable by 3");
```
Comment (RazvanN7) [inline std/bitmanip.d:289]:
```suggestion
                  "Wrong number of arguments (" ~ to!string(T.length) ~ "): Must be a multiple of 3");
```

**Review by RazvanN7 — APPROVED** [review]
Mind the nit.

**Comment by CyberShadow** [comment]
This pull request introduced a regression:
https://issues.dlang.org/show_bug.cgi?id=21725

**Comment by berni44** [comment]
> This pull request introduced a regression:
> https://issues.dlang.org/show_bug.cgi?id=21725

Thanks for fixing it!

**Pull request merged**